packages/discovery-provider/src/queries/get_collection_library.py

```diff
@@ -215,6 +215,52 @@ def _get_collection_library(args: GetCollectionLibraryArgs, session):
         user_id,
     )
 
+    # exclude hidden playlists and hidden albums which were not previously purchase by current user
+    playlists = list(
+        filter(
+            lambda playlist: not playlist["is_private"]
+            or (
+                collection_type == CollectionType.album
+                and playlist["stream_conditions"]
+                and "usdc_purchase" in playlist["stream_conditions"]
+                and playlist["access"]["stream"]
```

Comment:
I don't think this works either? This filters for purchased content _that is still purchase gated_